Source/JavaScriptCore/runtime/DeferredWorkTimer.cpp

```diff
@@ -212,6 +212,24 @@ bool DeferredWorkTimer::cancelPendingWork(Ticket ticket)
     return result;
 }
 
+bool DeferredWorkTimer::cancelAndRemovePendingWork(Ticket ticket)
+{
+    ASSERT(m_pendingTickets.contains(ticket));
+    ASSERT(ticket->isCancelled() || ticket->vm().currentThreadIsHoldingAPILock() || (Thread::mayBeGCThread() && ticket->vm().heap.worldIsStopped()));
+
+    bool result = false;
+    if (!ticket->isCancelled()) {
+        dataLogLnIf(DeferredWorkTimerInternal::verbose, "Canceling and removing ticket: ", RawPointer(ticket));
+        ticket->cancel();
+        {
+            Locker locker { m_taskLock };
+            result = m_pendingTickets.remove(ticket);
+        }
+    }
+
+    return result;
+}
```

Comment:
So, the question is why is it OK to be called from the different thread?

Reply:
The purpose of creating and using `DeferredWorkTimer::cancelAndRemovePendingWork` is to let the vm hanging on the deferred work which never be scheduled.

Reply:
I think this does not work when it is called from different thread.
Did you check `ticket->cancel` implementation?